src/metabase/query_processor/middleware/add_implicit_clauses.clj

```diff
@@ -125,14 +134,17 @@
 
 (defn add-implicit-mbql-clauses
   "Add implicit clauses such as `:fields` and `:order-by` to an 'inner' MBQL query as needed."
-  [{:keys [source-query], :as inner-query}]
-  (let [mbql-source-query? (and source-query (not (:native source-query)))
-        inner-query        (-> inner-query add-implicit-breakout-order-by add-implicit-fields)]
-    (if mbql-source-query?
-      ;; if query has an MBQL source query recursively add implicit clauses to that too as needed
-      (update inner-query :source-query add-implicit-mbql-clauses)
-      ;; otherwise we're done
-      inner-query)))
+  [form]
+  (walk/postwalk
+   (fn [form]
+     ;; add implicit clauses to any 'inner query', except for joins themselves (we should still add implicit clauses
+     ;; like `:fields` to source queries *inside* joins)
+     (if (and (map? form)
```

Comment:
it seems that this postwalk will just shove the implicit clauses even into the native queries now? was that intended?

Reply:
see the line below: it only transforms maps that have `:source-table` or `:source-query` in them -- that's how we know it's an MBQL query and not a native query